site/src/pages/WorkspacesPage/WorkspacesButton.tsx

```diff
@@ -152,75 +62,186 @@ export function WorkspacesButton({
   }
 
   return (
-    <PopoverContainer
-      // Stopgap value until bug where string-based horizontal origin isn't
-      // being applied consistently can get figured out
-      originX={-115}
-      originY="bottom"
-      sx={{ display: "flex", flexFlow: "column nowrap" }}
-      anchorButton={
-        <Button startIcon={<AddIcon />} variant="contained">
-          {children}
-        </Button>
-      }
-    >
-      <SearchBox
-        value={searchTerm}
-        onValueChange={(newValue) => setSearchTerm(newValue)}
-        placeholder="Type/select a workspace template"
-        label="Template select for workspace"
-        sx={{ flexShrink: 0, columnGap: COLUMN_GAP }}
-      />
-
-      <OverflowY
-        maxHeight={380}
-        sx={{
-          display: "flex",
-          flexFlow: "column nowrap",
-          paddingY: 1,
+    <>
+      <Button
+        startIcon={<AddIcon />}
+        variant="contained"
+        ref={anchorRef}
+        onClick={() => {
+          setIsOpen(true);
         }}
       >
-        {templatesFetchStatus === "loading" ? (
-          <Loader size={14} />
-        ) : (
-          <>
-            {processed.map((template) => (
-              <WorkspaceResultsRow key={template.id} template={template} />
-            ))}
-
-            {emptyState}
-          </>
-        )}
-      </OverflowY>
-
-      <Link
-        component={RouterLink}
-        to="/templates"
-        sx={{
-          outline: "none",
-          "&:focus": {
-            backgroundColor: theme.palette.action.focus,
-          },
+        {children}
+      </Button>
+      <Popover
+        disablePortal
+        open={isOpen}
+        onClose={() => setIsOpen(false)}
+        anchorEl={anchorRef.current}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "right",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "right",
         }}
+        css={(theme) => ({
+          marginTop: theme.spacing(1),
```

Comment:
What are your thoughts on replacing this with padding (probably from a parent element), just to avoid any risks of margin collapse in the future?

Reply:
I see, I think it makes sense.